### ISSUE TITLE
Suggestion: dropdown list

### ISSUE THREAD

**Issue by luukatyuo** [speedball2001/identitychooser-mx#63]
Hi there,
i ve been a fan of this awesome extension since forever. It never failed me ;-)
I guess since its been reimplemented as a MailExtension there also has been a UI-"upgrade", but sadly to me it's kills ma workflow.
In older versions there ve been an drop-down-menue, when u clicked the write option where could choose which mailadress to pick as sender. 
I'd rly love to see this or something like that again if it's possible to do. (Maybe as an option? coz some ppl might pref. the new UI-version of choosing the sender?)

**Comment by speedball2001** [comment]
I understand the sentiment, in fact I liked the old way of doing it as well. But with the move to Mail Extension there's simply no option of continuing doing it the old way. First of all, Thunderbird does not want us to modify UI elements anymore. So that is not an option anymore. Secondly, there's no way of figuring out the coordinates of buttons to make windows appear below a button.

I thought about all of this a lot and it was a decision between the new style or discontinuing development of Identity Chooser completely. From my personal use I can say that adapting to the new style took me maybe a week. It's different from the old way, but the number of clicks stays the same. On top of that the new style will always work, even if you create a new email via different means, i.e. when you click on a mailto:-link or via the address book.

What we could do is trying to improve the window with the identities and I appreciate mockups from users with improved US as well as help with implementation.

**Comment by bvandergiessen** [comment]
Is this not possible with a BrowserAction, i.e. a toolbar button that belongs to the extension, which the user could put in place of (or next to) the built-in 'Create a new message' button?

A BrowserAction is allowed to show a popup menu when clicked. You can also program it to show the accounts list only on a middle-button-click while choosing the default account on a left-click, or the other way around, or configurable.

See https://webextension-api.thunderbird.net/en/stable/browserAction.html

**Comment by speedball2001** [comment]
> Is this not possible with a BrowserAction, i.e. a toolbar button that belongs to the extension, which the user could put in place of (or next to) the built-in 'Create a new message' button?
> 
> A BrowserAction is allowed to show a popup menu when clicked. You can also program it to show the accounts list only on a middle-button-click while choosing the default account on a left-click, or the other way around, or configurable.
> 
> See https://webextension-api.thunderbird.net/en/stable/browserAction.html

This is true but would only solve the issue when creating a new email using the "Create New Message" button in the main window. But there are a lot of other ways to create a new email: When clicking on the menu File->New->Message; when clicking on a mailto: link in an email; when a web site let's you create an email in your local email program; when emailing someone from the address book, and so on. I can't hook into any of these with a mail extension except using the new way that shows the popup after the editor window is created by Thunderbird. 

Furthermore a lot of users do use Identity Chooser for forwarding and replying. Since a Mail Extension can have only one browserAction (and only one messageDisplayAction) this wouldn't work for these invocations.

That's why Identity Chooser works as it does. As I already said elsewhere: I do appreciate mockups and/or pull requests that improve the styling of the popup and the accounts list. But I wont change the fundamental way the extension works.

**Comment by bvandergiessen** [comment]
> But there are a lot of other ways to create a new email: When clicking on the menu File->New->Message; when clicking on a mailto: link in an email; when a web site let's you create an email in your local email program; when emailing someone from the address book, and so on.

I don't remember how the old version would deal with those situations. Presumably not by opening the dropdown on the 'new mail' button, right?

**Comment by speedball2001** [comment]
The old version did not work in these situations. Identity Chooser 4.x improves this a lot.

**Comment by bvandergiessen** [comment]
> As I already said elsewhere: I do appreciate mockups and/or pull requests that improve the styling of the popup and the accounts list.

Not exactly a styling suggestion but, I guess, "UX": when you click Cancel (or you close the Identity Chooser popup by clicking the OS's window-closing icon), it also closes the new-mail compose window. It would be better if it left that open, so you can write the email from the default sender account.

In the end, Identity Chooser is a popup above another window, and the expected behavior is that closing a popup only closes that popup. When the user thinks "I don't want to think about my accounts just now, I'll just send the email", that is a decision that's up to the user.

**Comment by oleole39** [comment]
Hello and thanks for your all your work so far on Identity Chooser.

I recently updated to TB 115 and have the unpleasant surprise on IC's new behavior. In my case things are even worse as when I click on the New Message button on the main TB window, comes on the foreground the message editor and on the background the IC window. Consequently IC window unfortunately doesn't bring much. And it has sometimes annoyed me, as I start writing the message, the shift windows to check something here or there and happen to fall on the IC window, click "cancel" and it closes the message editor window I had started to write in without saving.

However my particular issue is not only due to IC but also to some not-obvious-to-fix bug in my OS which for some reasons often open new windows on the background (whether TB's or other softwares'). I didn't have the issue with the dropdown (and would have favored the dropdown option anyway), but reading all of the above I understand there is little hope to get it back. 

Considering my OS-related issue, I currently find IC more annoying than convenient. Just sharing an user experience that might belong to the edge ones.

**Comment by oleole39** [comment]
Just a slight step forward in the hope of restoring IC UI as a dropdown menu under the "Write Message" button.

I noted several extensions can mimic the tag button at the top of the message pane, which displays a proper dropdown menu, such as:
- [ImportExport Tools NG](https://github.com/thunderbird/import-export-tools-ng); I assume that the menu is created with helpers defined [in that file](https://github.com/thunderbird/import-export-tools-ng/blob/master/src/wextMenus.js), referring to the menu name `msgCtxMenuSet`
- [Printing Tools NG](https://github.com/cleidigh/printing-tools-ng)

![TestIC](https://github.com/speedball2001/identitychooser-mx/assets/59071673/ad57049c-848a-4dca-ba1f-84772c9c9fb5)

To get above display I used the GUI and:
1. Hid the folder pan header (3 dots button => hide folder pane header)
2. Customized the unified toolbar to move ImportExport Tools NG's toolbar button nearby the "Write" button, I could get something close to the expected result. We could imagine a custom toolbar button created by IC which would replace the "Write" toolbar button (identified by CSS selector `li.write-message`)  

This would not yet display a dropdown list under the "New Message" button (identified by CSS Selector `#folderPaneWriteMessage`) which was hidden here together with the whole folder pan header, but would already look much more convenient to me than the current popup.

**Comment by oleole39** [comment]
@speedball2001 Oh nice! Is it closed as completed although not released yet? or as not planned?

**Comment by speedball2001** [comment]
It's not something I'll be working on, therefore I closed this issue.
